apps/accounts/views.py

```diff
@@ -132,3 +135,16 @@ class ProfileView(JWTProtectedView, generics.RetrieveAPIView):
 
     def retrieve(self, request, *args, **kw):
         return Response(self.get_serializer(request.user).data)
+
+
+class AccountSourceView(generics.RetrieveAPIView):
+    serializer_class = AccountSourceSerializer
+
+    @waffle_switch('fxa-auth')
+    def retrieve(self, request, *args, **kwargs):
+        email = request.GET.get('email')
+        try:
+            user = UserProfile.objects.get(email=email)
+        except UserProfile.DoesNotExist:
```

Comment:
I like putting the logic here a lot better, thanks. I think it could still use a comment here so that it's clear that `fxa` isn't really the source but a way to obscure non-existing accounts.